frontend_tests/node_tests/composebox_typeahead.js

```diff
@@ -96,60 +96,70 @@ const emoji_stadium = {
     aliases: ["stadium"],
     emoji_url: "TBD",
     emoji_code: "1f3df",
+    reaction_type: "unicode_emoji",
 };
 const emoji_tada = {
     name: "tada",
     aliases: ["tada"],
     emoji_url: "TBD",
     emoji_code: "1f389",
+    reaction_type: "unicode_emoji",
 };
 const emoji_moneybag = {
     name: "moneybag",
     aliases: ["moneybag"],
     emoji_url: "TBD",
     emoji_code: "1f4b0",
+    reaction_type: "unicode_emoji",
 };
 const emoji_japanese_post_office = {
     name: "japanese_post_office",
     aliases: ["japanese_post_office"],
     emoji_url: "TBD",
     emoji_code: "1f3e3",
+    reaction_type: "unicode_emoji",
 };
 const emoji_panda_face = {
     name: "panda_face",
     aliases: ["panda_face"],
     emoji_url: "TBD",
     emoji_code: "1f43c",
+    reaction_type: "unicode_emoji",
 };
 const emoji_see_no_evil = {
     name: "see_no_evil",
     aliases: ["see_no_evil"],
     emoji_url: "TBD",
     emoji_code: "1f648",
+    reaction_type: "unicode_emoji",
 };
 const emoji_thumbs_up = {
     name: "thumbs_up",
     aliases: ["thumbs_up"],
     emoji_url: "TBD",
     emoji_code: "1f44d",
+    reaction_type: "unicode_emoji",
 };
 const emoji_thermometer = {
     name: "thermometer",
     aliases: ["thermometer"],
     emoji_url: "TBD",
     emoji_code: "1f321",
+    reaction_type: "unicode_emoji",
```

Comment:
It would be quite be helpful to split these test changes out as a preparatory commit with its own commit message. IF I understand correctly, these node tests previously had invalid setup (not using the true type for emoji data), and while fixing that is an important part of the change you're making, it'd be much more readable to have it be its own change with its own explanatory commit message, since this part has no functional changes.

Check out the Zulip commit guidelines for more details: https://zulip.readthedocs.io/en/latest/contributing/version-control.html

Reply:
@timabbott I'm a little confused here. I was particularly following the commit guidelines by adding the test fix in the same commit as the implementation:

> It should pass tests (so test updates needed by a change should be in the same commit as the original change, not a separate “fix the tests that were broken by the last commit” commit).

Is there something I'm missing? Is there a section in the guideline that talks about "preparatory commits"; could you please point me to it?

Reply:
Yeah, I think I'd misunderstood the code a bit. I guess what makes sense here is to have a preparatory commit that just starts passing the `reaction_type` around (and has these test setup changes), but doesn't have the function change of using it. Does that makse sense?